docs/redux.md

```diff
@@ -0,0 +1,118 @@
+needs to be able to render several different layouts (pages) depending on user interaction. State
+management is essential for this process. FTW needs to know if a user has been authenticated, if it
+has received relevant data for the current page, and so on.
+
+We use [Redux](https://redux.js.org/introduction) for state management on the application level. You
+should read more about Redux before you start modifying queries to Flex API or creating new Page
+level elements (unless you are modifying [a static page](static-pages.md)).
+
+In the following subtopics, we assume that you know the
+[basics of Redux](https://redux.js.org/basics) already.
+
+* [Containers: Pages + TopbarContainer](#containers-pages--topbarcontainer)
+* [Duck files](#duck-files)
+* [Setting up Redux](#setting-up-redux)
+* [Advanced Redux concepts: thunks](#advanced-redux-concepts-thunks)
+
+## Containers: Pages + TopbarContainer
+
+We have set up FTW so that pages are aware of Redux, but other components and forms are purely
+presentational components. This makes it easier to customize UI components - you don't need to be
+aware of the complexity related to Redux setup when you just want to touch the behavior of some
+visual component. In those cases, you can just head to `scr/components/` or `src/forms/` folder and
+you can see from there what props are available for each component when they are rendered.
+
+Naturally, there is a need for higher level components which fetch new data and define what props
+are passed down to presentational components. In Redux terminology, those components are called
+_Containers_. FTW has defined all the containers inside a folder called `src/containers/`. It
+contains all the pages and a special container for top bar (TopbarContainer) since that is a very
+complex component and it is shared with almost every page.
+
+The actual container setup of a page level component can be found from the bottom of the component
+file. For example, `src/containers/TransactionPage/TransactionPage.js` connects itself to Redux
+store with `mapStateToProps` and `mapDispatchToProps` functions:
+
+```js
+const TransactionPage = compose(connect(mapStateToProps, mapDispatchToProps), injectIntl)(
+  TransactionPageComponent
+);
+```
+
+## Duck files
+
+Inside `src/containers/<ComponentName>` folder, we have also a special `<ComponentName>.duck.js`
+file. These files wrap all the other page-specific Redux concepts into a single file. Instead of
+writing action types, action creators and reducers in separate files (and spread them around our
+folder structure), FTW tries to keep pages isolated. Page specific actions, store updates, and data
+fetches happen inside their respective folder - just look for a file which name follows a pattern:
+`<ComponentName>.duck.js`. This pattern was first proposed by Erik Rasmussen and you can read more
+about it from [his repository](https://github.com/erikras/ducks-modular-redux).
```

Comment:
Some suggestions:

- folder -> directory
- isolated -> encapsulated

Maybe add a sentence that ducks are really a simple concept of just adding related things into a single file.

Reply:
Changed. (even though Rasmussen uses "isolated")